components/script/script_runtime.rs

```diff
@@ -125,6 +136,73 @@ unsafe extern "C" fn enqueue_job(cx: *mut JSContext,
     }), false)
 }
 
+#[allow(unsafe_code)]
+/// https://html.spec.whatwg.org/multipage/#the-hostpromiserejectiontracker-implementation
+unsafe extern "C" fn promise_rejection_tracker(
+    cx: *mut JSContext,
+    promise: HandleObject,
+    state: PromiseRejectionHandlingState,
+    _data: *mut c_void
+) {
+    let global = GlobalScope::from_context(cx);
+
+    wrap_panic(AssertUnwindSafe(|| {
+        match state {
+            PromiseRejectionHandlingState::Unhandled => {
+                global.add_uncaught_rejection(promise);
+            },
+            PromiseRejectionHandlingState::Handled => {
+                global.add_consumed_rejection(promise);
+            }
+        };
+    }), ());
+}
+
+#[allow(unsafe_code, unrooted_must_root)]
+/// https://html.spec.whatwg.org/multipage/#notify-about-rejected-promises
+pub unsafe fn notify_about_rejected_promises(global: DomRoot<GlobalScope>) {
+    let cx = global.get_cx();
+
+    let uncaught_rejections: Vec<TrustedPromise> = global.get_uncaught_rejections()
+        .borrow()
+        .iter()
+        .map(|promise| {
+            let promise = Promise::new_with_js_promise(Handle::from_raw(promise.handle()), cx);
+
+            TrustedPromise::new(promise)
+        })
+        .collect();
+
+    let target = Trusted::new(global.upcast::<EventTarget>());
+
+    global.as_window().dom_manipulation_task_source().queue(
```

Comment:
A bunch of worker tests panicked because the global isn't a window. We're going to need to expose `dom_manipulation_task_source()` on GlobalScope and ensure that worker globals can use it as well. That should just mean moving [this](https://github.com/servo/servo/blob/master/components/script/dom/window.rs#L162) to GlobalScope and creating it [here](https://github.com/servo/servo/blob/master/components/script/dom/workerglobalscope.rs#L108) to pass to GlobalScope::new_inherited.